### PR TITLE
Return attention mask in ASR pipeline to avoid warnings

### LINKED ISSUE #33498: The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
### System Info

- `transformers` version: 4.44.2
- Platform: Linux-6.8.0-44-generic-x86_64-with-glibc2.39
- Python version: 3.12.3
- Huggingface_hub version: 0.24.7
- Safetensors version: 0.4.5
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.4.1+cu121 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No

### Who can help?

speech models: @ylacombe, @eustlb
pipelines: @Rocketknight1

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
import torch 
from transformers import pipeline

pipe = pipeline(
    "automatic-speech-recognition",
    model="openai/whisper-base.en",
    device="cpu",
    torch_dtype=torch.float32,
)

# https://github.com/openai/whisper/blob/main/tests/jfk.flac
pipe("./jfk.flac")
```

### Expected behavior

This does return the expected:

```python
{'text': ' And so my fellow Americans ask not what your country can do for you, ask what you can do for your country.'}
```

But it also prints the following, so would be nice to fix/suppress:

```
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
```

Thanks!

### PR DESCRIPTION
**Pull request by Rocketknight1**
cc @sanchit-gandhi - this PR just sets `return_attention_mask=True` on the preprocessors in the `automatic_speech_recognition` pipeline to avoid warnings caused by missing attention masks. It works okay in my testing, but please let me know if you think it could cause any problems!

Fixes #33498

### CODE REVIEW

**Comment by Vaibhavs10** [comment]
cc: @ylacombe here

**Comment by ylacombe** [comment]
Hey @Rocketknight1, thanks for opening the PR!
In theory, I don't see any problems with this fix, have you been able to run every slow ASR pipeline tests here ?

**Comment by Rocketknight1** [comment]
@ylacombe there are some slow tests that I can't get working on my local machine (even on `main`). However, of the tests that run, they all work with this PR as well!

**Comment by ylacombe** [comment]
Let me know if you want me to run them!

**Comment by Rocketknight1** [comment]
@ylacombe Sure!

**Review by ylacombe — APPROVED** [review]
So, I've first opened a PR to fix some of the slow tests that were not passing due to how data was loaded #33545. 
Your PR doesn't add any failing tests as compared to main and the changes make sense, so I think it should be ok to merge !

**Review by LysandreJik — APPROVED** [review]
Thanks for the PR @Rocketknight1!

**Comment by monica-sekoyan** [comment]
Hi @Rocketknight1, 
I think `return_attention_mask` should also be added to `chunk_iter` function, so the warning will be removed even when we specify `chunk_length_s`.

**Comment by Holmes-GU** [comment]
> cc @sanchit-gandhi - this PR just sets `return_attention_mask=True` on the preprocessors in the `automatic_speech_recognition` pipeline to avoid warnings caused by missing attention masks. It works okay in my testing, but please let me know if you think it could cause any problems!
> 
> Fixes #33498

Hi, where can I set `return_attention_mask=True`?

**Comment by AlessandroSpallina** [comment]
any update on this?

**Comment by AlessandroSpallina** [comment]
I'm having the error fixed here using latest transformers==4.47.1:
```
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Whisper did not predict an ending timestamp, which can happen if audio is cut off in the middle of a word. Also make sure WhisperTimeStampLogitsProcessor was used during generation.
```

when running this code:
```
        pipe = pipeline(
            "automatic-speech-recognition",
            model=self.model,
            torch_dtype=torch.float16,
            chunk_length_s=30,
            batch_size=24,
            return_timestamps=True,
            device=self.device,
            tokenizer=processor.tokenizer,
            feature_extractor=processor.feature_extractor,
            model_kwargs={"use_flash_attention_2": True},
            generate_kwargs={
                "max_new_tokens": 128,
            },
        )
```
